test/extend_mongodb_tests.js

```diff
@@ -158,4 +158,94 @@ describe('Extended JSON', function() {
     );
     expect(parsedRegExp).to.eql(parsedBSONRegExp);
   });
+
+  it('should serialize from BSON object to EJSON object', function() {
+    const doc = {
+      binary: new Binary(''),
+      code: new Code('function() {}'),
+      dbRef: new DBRef('tests', new Int32(1), 'test'),
+      decimal128: new Decimal128(128),
+      double: new Double(10.1),
+      int32: new Int32(10),
+      long: new Long(234),
+      maxKey: new MaxKey(),
+      minKey: new MinKey(),
+      objectID: ObjectID.createFromHexString('111111111111111111111111'),
+      bsonRegExp: new BSONRegExp('hello world', 'i'),
+      symbol: new Symbol('symbol'),
+      timestamp: new Timestamp()
+    };
+
+    const result = extJSON.serialize(doc);
+
+    expect(result).to.deep.equal({
+      binary: { $binary: { base64: '', subType: '00' } },
+      code: { $code: 'function() {}' },
+      dbRef: { $ref: 'tests', $id: { $numberInt: '1' }, $db: 'test' },
+      decimal128: { $numberDecimal: '0E-6176' },
+      double: { $numberDouble: '10.1' },
+      int32: { $numberInt: '10' },
+      long: { $numberLong: '234' },
+      maxKey: { $maxKey: 1 },
+      minKey: { $minKey: 1 },
+      objectID: { $oid: '111111111111111111111111' },
+      bsonRegExp: { $regularExpression: { pattern: 'hello world', options: 'i' } },
+      symbol: { $symbol: 'symbol' },
+      timestamp: { $timestamp: { t: 0, i: 0 } }
+    });
+  });
+
+  it('should deserialize from EJSON object to BSON object', function() {
+    const doc = {
+      binary: { $binary: { base64: '', subType: '00' } },
+      code: { $code: 'function() {}' },
+      dbRef: { $ref: 'tests', $id: { $numberInt: '1' }, $db: 'test' },
+      decimal128: { $numberDecimal: '0E-6176' },
+      double: { $numberDouble: '10.1' },
+      int32: { $numberInt: '10' },
+      long: { $numberLong: '234' },
+      maxKey: { $maxKey: 1 },
+      minKey: { $minKey: 1 },
+      objectID: { $oid: '111111111111111111111111' },
+      bsonRegExp: { $regularExpression: { pattern: 'hello world', options: 'i' } },
+      symbol: { $symbol: 'symbol' },
+      timestamp: { $timestamp: { t: 0, i: 0 } }
+    };
+
+    const result = extJSON.deserialize(doc);
+
+    // binary
+    expect(result.binary).to.be.an.instanceOf(BSON.Binary);
```

Comment:
Not worth rewriting, but for more verbose / detailed assertion errors, you can do:

```js
expect(result).to.have.property('binary').that.is.an.instanceOf(BSON.Binary);
```

This will give you more detailed errors about what exactly is wrong.